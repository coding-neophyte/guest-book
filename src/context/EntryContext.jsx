import { useState, useContext, createContext } from 'react'

const EntryContext = createContext();

const EntryProvider = ({ children }) => {
    const [entry, setEntry] = useState([]);

    return <EntryContext.Provider value={{entry, setEntry}}>
        {children}
    </EntryContext.Provider>
}

const useEntry = () =>{
    const context= useContext(EntryContext);

    if(context === undefined){
        throw new Error('invalid provider')
    }
    return context
}

export {EntryProvider, useEntry}
