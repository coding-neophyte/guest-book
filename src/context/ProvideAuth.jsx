import { createContext, useState } from "react";


const AuthCtx = createContext();

const AuthProvider = ({ children }) => {
    const [user, setUser] = useState()

    const login = (username, password) => {
      const loginVerified =  username === process.env.AUTH_USERNAME && password === process.env.AUTH_PASSWORD;

      if(loginVerified){
          setUser(username)
      }
      return loginVerified;
    }

    const logout = () => {
        setUser(null)
    }

    return (
        <AuthCtx.Provider value={{ user, login, logout }}>{children}</AuthCtx.Provider>
    );
}

export { AuthCtx, AuthProvider }
