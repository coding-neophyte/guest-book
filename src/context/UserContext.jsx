import { useState, useContext, createContext } from 'react'

const UserContext = createContext()

const UserProvider = ({ children }) => {
    const [user, setUser] = useState({})

    return <UserContext.Provider value={{user, setUser}}>
        {children}
    </UserContext.Provider>
}


const UseUser = () => {
    const context = useContext(context)

    if(context === undefined){
        throw new Error('invalid provider')
    }
    return context 
}

export { UserProvider, UseUser }
