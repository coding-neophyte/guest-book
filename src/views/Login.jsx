import React from 'react'
import useAuth from '../hooks/useAuth'
import { useHistory, useLocation } from 'react-router-dom'
import { useState } from 'react'

export default function Login() {
    let history = useHistory()
    let location = useLocation()
    let auth = useAuth()
    let { from } = location.state || { from: { pathname: '/' }}

   const [error, setError] = useState(null)
   const [username, setUsername] = useState('')
   const [password, setPassword] = useState('')

    const handleLogin = (e) => {
        e.preventDefault();

        const loginSuccess = auth.login(username, password)

       if (loginSuccess) {


        history.replace(from)
        } else {
            setError('Incorrect Username or Password')
        }

    }

    return (
        <div>
            <form onSubmit={handleLogin}>
                <fieldset>
                    <legend>Sign In</legend>
                    <label htmlFor='username'>Username</label>
                    <input type='text' id='username' name='username' value={username} onChange={(e) => setUsername(e.target.value)}/>

                    <label htmlFor='password'>Password</label>
                    <input type='password' id='password' name='password' value={password} onChange={(e) => setPassword(e.target.value)}/>
                </fieldset>
                    <button>Signin</button>
            </form>
            {error && <h4>{error}</h4>}
        </div>
    )
}
