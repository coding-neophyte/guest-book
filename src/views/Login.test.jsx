import { screen, render } from '@testing-library/react'
import Login from './Login'
import { MemoryRouter } from 'react-router-dom'
import { AuthProvider } from '../context/ProvideAuth'



it.only('should render login page', () => {
    const { container } = render(
    <MemoryRouter initialEntries={['/login']}>
        <AuthProvider>
        <Login />
        </AuthProvider>
    </MemoryRouter>)


    expect(container).toMatchSnapshot();
})
