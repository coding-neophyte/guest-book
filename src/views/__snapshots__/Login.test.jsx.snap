// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login page 1`] = `
<div>
  <div>
    <form>
      <fieldset>
        <legend>
          Sign In
        </legend>
        <label
          for="username"
        >
          Username
        </label>
        <input
          id="username"
          name="username"
          type="text"
          value=""
        />
        <label
          for="password"
        >
          Password
        </label>
        <input
          id="password"
          name="password"
          type="password"
          value=""
        />
      </fieldset>
      <button>
        Signin
      </button>
    </form>
  </div>
</div>
`;
